src/java/org/apache/cassandra/db/compression/CompressionDictionaryDetailsTabularData.java

```diff
@@ -0,0 +1,229 @@
+                                            pojo.dictChecksum,
+                                            pojo.dictLength
+                                            });
+        }
+        catch (OpenDataException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Deserializes data to convenience object to work further with.
+     *
+     * @param compositeData data to create pojo from
+     * @return deserialized composite data to convenience object
+     * @throws IllegalArgumentException if values in deserialized object are invalid.
+     * @see CompressionDictionaryPojo#validate()
+     */
+    public static CompressionDictionaryPojo from(CompositeData compositeData)
+    {
+        String keyspace = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KEYSPACE_NAME);
+        String table = (String) compositeData.get(CompressionDictionaryDetailsTabularData.TABLE_NAME);
+        long dictId = (Long) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_ID_NAME);
+        byte[] dictionaryBytes = (byte[]) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_NAME);
+        String kind = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KIND_NAME);
+        int checksum = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.CHECKSUM_NAME);
+        int size = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.SIZE_NAME);
+
+        CompressionDictionaryPojo pojo = new CompressionDictionaryPojo();
+        pojo.keyspace = keyspace;
+        pojo.table = table;
+        pojo.dictId = dictId;
+        pojo.dict = dictionaryBytes;
+        pojo.kind = kind;
+        pojo.dictChecksum = checksum;
+        pojo.dictLength = size;
+
+        pojo.validate();
+
+        return pojo;
+    }
+
+    public static class CompressionDictionaryPojo
+    {
+        public String keyspace;
+        public String table;
+        public long dictId;
+        public byte[] dict;
+        public String kind;
+        public int dictChecksum;
+        public int dictLength;
+
+        /**
+         * Dictionary is valid if, keyspace and table are specified, dictionary id is strictly positive integer,
+         * dictionary byte array is not nor not empty,
+         * kind corresponds to {@code Kind}, checksum and size are bigger than 0.
+         */
+        public void validate()
+        {
+            if (keyspace == null)
+                throw new IllegalArgumentException("keyspace not specified");
+            if (table == null)
+                throw new IllegalArgumentException("table not specified");
+            if (dictId <= 0)
+                throw new IllegalArgumentException("Provided dictionary id is lower than 0, it is '" + dictId + "'.'");
+            if (dict == null || dict.length == 0)
+                throw new IllegalArgumentException("Provided dictionary byte array is null or empty.");
```

Comment:
Can we add a new check to ensure the importing dictionary size does not exceed certain amount? We will not want to import dictionary that is larger than 1MiB, most likely the dictionary should be less than 100KiB.